### PR TITLE
Reimplement Flex Layout According to Spec

### PR DESCRIPTION
**Pull request by lkzhao**
Added support for more justify content and fixed a few bugs with flex layout.

### CODE REVIEW

File: Sources/Layout/FlexLayout.swift [hunk 1]
```diff
@@ -50,46 +50,39 @@ public class FlexLayout<Data>: AxisDependentLayout<Data> {
...
+    let (sizes, totalPrimary) = getSizes(collectionSize: collectionSize,
+                                         dataProvider: dataProvider,
+                                         sizeProvider: sizeProvider)
 
     var offset: CGFloat = 0
-    if totalHeight < primary(collectionSize), totalFlex == 0 {
+    var padding = self.padding
+    if totalPrimary < primary(collectionSize) {
+      let leftOverPrimaryWithPadding = primary(collectionSize) - totalPrimary
+      let leftOverPrimary = leftOverPrimaryWithPadding + self.padding * CGFloat(dataProvider.numberOfItems - 1)
```
Comment (jindulys) [inline Sources/Layout/FlexLayout.swift]: leftOverPrimary = leftOverPrimaryWithPadding - self.padding * CGFloat(dataProvider.numberOfItems - 1)?

Comment (lkzhao) [reply Sources/Layout/FlexLayout.swift]: TotalPrimary is calculated from all cell height + all minimum paddings inbetween. Therefore if we want to remove the paddings from our calculation we need to subtract it by (count - 1) * minimum padding

Comment (jindulys) [reply Sources/Layout/FlexLayout.swift]: gotcha

**Comment by jindulys** [comment]
+1

**Comment by jindulys** [comment]
+1

**Pull request merged**